### PR TITLE
Bf/bucket parsing

### PR DESCRIPTION
**Pull request by LaurenSpiegel**
This closes https://github.com/scality/S3/issues/157

### CODE REVIEW

**Comment by LaurenSpiegel** [comment]
@ironman-machine check it out

File: lib/utils.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ utils.getNamesFromReq = function getNamesFromReq(request, bucketNameFromHost) {
             resources.bucket = urlArr[1];
             resources.object = urlArr.slice(2).join('/');
         } else if (urlArr.length === 1) {
-            resources.bucket = urlArr[1];
+            resources.bucket = urlArr[0];
```
Comment (nicolas2bert) [inline lib/utils.js]: Why have you changed that? 
For exemple if path == '/mybucket'  so, resources.bucket == ''
No?

**Comment by nicolas2bert** [comment]
👍

**Comment by rahulreddy** [comment]
👍

**Pull request merged**